### DISCUSSION TITLE
Should we use deserialize(serialize(x)) to clone x?

### DISCUSSION THREAD

**Discussion by geoknee** [statechannels/go-nitro#413] [Ideas]
Instead of our existing `.clone()` methods?

**Comment by lalexgap** [comment]
I started looking into using reflection to make sure clone doesn't have a shallow copy [here](https://github.com/statechannels/go-nitro/pull/419). However it's already eaten up a bunch of time and would probably require its own set of tests so we could trust it fully.

I think switching to `deserialize(serialize(x))` would be much more straightforward and quicker than using reflection so that would be my vote.
